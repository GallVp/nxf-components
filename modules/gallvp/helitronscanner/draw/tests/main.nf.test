nextflow_process {

    name "Test Process HELITRONSCANNER_SCANHEAD"
    script "../main.nf"
    process "HELITRONSCANNER_SCANHEAD"

    tag "modules"
    tag "modules_gallvp"
    tag "helitronscanner"
    tag "helitronscanner/scanhead"
    tag "gunzip"

    setup {
        run('GUNZIP') {
            script "../../../gunzip/main.nf"

            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/eukaryotes/actinidia_chinensis/genome/chr1/genome.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("actinidia_chinensis-genome_1_fasta_gz-success") {

        when {
            process {
                """
                input[0] = GUNZIP.out.gunzip
                input[1] = []
                input[2] = 0
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.head[0][1]).text.contains('2729827:1 2729980:1 2730005:1') },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

    test("sarscov2 - genome - success") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = []
                input[2] = 0
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.head[0][1]).text.contains('113:2 236:1 1158:1 1754:2 1771:1') },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

    test("stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = GUNZIP.out.gunzip
                input[1] = []
                input[2] = 0
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
