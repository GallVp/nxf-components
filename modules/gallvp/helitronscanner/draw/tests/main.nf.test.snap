{
    "actinidia_chinensis-genome_1_fasta_gz-success": {
        "content": [
            [
                "versions.yml:md5,d5784b052afc56876f5e4d7eb83f518d"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-03T15:05:25.150055"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.head:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d5784b052afc56876f5e4d7eb83f518d"
                ],
                "head": [
                    [
                        {
                            "id": "test"
                        },
                        "test.head:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d5784b052afc56876f5e4d7eb83f518d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-03T15:04:14.669587"
    },
    "sarscov2 - genome - success": {
        "content": [
            [
                "versions.yml:md5,d5784b052afc56876f5e4d7eb83f518d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-25T12:57:11.212315"
    }
}